viewer/src/migration/Cognite3DViewer.ts

```diff
@@ -0,0 +1,448 @@
+import { CogniteClient } from '@cognite/sdk';
+import RenderController from './RenderController';
+
+export class Cognite3DViewer {
+  private get canvas(): HTMLCanvasElement {
+    return this.renderer.domElement;
+  }
+  static isBrowserSupported(): boolean {
+    return true;
+  }
+
+  readonly domElement: HTMLElement;
+  private readonly renderer: THREE.WebGLRenderer | Cognite3DThreeRenderer;
+  private readonly camera: THREE.PerspectiveCamera;
+  private readonly scene: THREE.Scene;
+  private readonly controls: ComboControls;
+  private readonly sdkClient: CogniteClient;
+
+  private readonly eventListeners = {
+    cameraChange: new Array<(pos: THREE.Vector3, target: THREE.Vector3) => void>()
+  };
+  private readonly models: Cognite3DModel[] = [];
+
+  private isDisposed = false;
+  private readonly forceRendering = false; // For future support
+
+  private readonly renderController: RenderController;
+  private latestRequestId: number = -1;
+  private readonly clock = new THREE.Clock();
+
+  constructor(options: Cognite3DViewerOptions) {
+    if (options.enableCache) {
+      throw new NotSupportedInMigrationWrapperError('Cache is not supported');
+    }
+    if (options.logMetrics) {
+      throw new NotSupportedInMigrationWrapperError('LogMetris is not supported');
+    }
+
+    this.renderer = options.renderer || new THREE.WebGLRenderer();
+    this.canvas.style.width = '640px';
+    this.canvas.style.height = '480px';
+    this.canvas.style.minWidth = '100%';
+    this.canvas.style.minHeight = '100%';
+    this.canvas.style.maxWidth = '100%';
+    this.canvas.style.maxHeight = '100%';
+    this.domElement = options.domElement || createCanvasWrapper();
+    this.domElement.appendChild(this.canvas);
+
+    this.camera = new THREE.PerspectiveCamera(60, undefined, 0.1, 10000);
+    this.camera.position.x = 30;
+    this.camera.position.y = 10;
+    this.camera.position.z = 50;
+    this.camera.lookAt(new THREE.Vector3());
+
+    this.scene = new THREE.Scene();
+    this.controls = new ComboControls(this.camera, this.canvas);
+    this.controls.dollyFactor = 0.992;
+    this.controls.addEventListener('cameraChange', event => {
+      const { position, target } = event.camera;
+      this.eventListeners.cameraChange.forEach(f => {
+        f(position.clone(), target.clone());
+      });
+    });
+
+    this.sdkClient = options.sdk;
+    this.renderController = new RenderController(this.camera);
+    this.animate(0);
+  }
+
+  dispose(): void {
+    if (this.isDisposed) {
+      return;
+    }
+
+    this.isDisposed = true;
+
+    if (this.latestRequestId !== undefined) {
+      cancelAnimationFrame(this.latestRequestId);
+    }
+
+    this.domElement.removeChild(this.canvas);
+    this.renderer.dispose();
+    this.scene.dispose();
+    while (this.models.length > 0) {
+      const model = this.models.pop()!;
+      model.dispose();
+    }
+  }
+
+  onClick(_callback: (event: PointerEvent) => void): void {
```

Comment:
Why not use the same API as the original viewer? Does this not introduce a migration to the migration? 😛